### PR TITLE
updates to batch and other codes to accommodate aperture guess, different std/target extraction options, and simple flux calibation

### PR DESCRIPTION
**Pull request by aburgasser**
implements mike's intermediate flux calibration; tested on multiple instrument settings

### CODE REVIEW

File: src/pyspextool/extract/extract_apertures.py [hunk 1]
```diff
@@ -508,7 +508,7 @@ def write_apertures(spectra, psbginfo=None, xsbginfo=None, optimal_info=None,
                     plot_spectra(output_fullpath + '.fits',flag_linearity=True,
                                  file_info=qafileinfo)
 
-        elif extract.state['reductionmode'] == 'A-Sky/Dark':
```
Comment (michaelcushing) [inline src/pyspextool/extract/extract_apertures.py:511]: Why the change?  Sky/Dark is the most generic version?  That is, if we want people to use it, then this mode is meant for either a Sky or dark frame.

File: src/pyspextool/extract/find_peaks.py [hunk 2]
```diff
@@ -131,8 +131,8 @@ def find_peaks(profiles, method_info, fwhm=0.8):
 
         for i in range(norders):
 
-            x = profiles[i]['y']
-            y = profiles[i]['p']
```
Comment (michaelcushing) [inline src/pyspextool/extract/find_peaks.py:135]: I don't see changes to make_profiles in the request so did you change the output of that program from 'y' and 'p' to 'angle' and 'profile'?

File: src/pyspextool/extract/find_peaks.py [hunk 3]
```diff
@@ -143,23 +143,25 @@ def find_peaks(profiles, method_info, fwhm=0.8):
...
 
             y = y - np.median(y)
+            absy = np.abs(y)
 
             # Now loop over each aperture and do the fit
 
             for j in range(npeaks):
                 # Get the peak value for the guess
 
-                idx = int(find_index(x, guess_peaks[i, j]))
```
Comment (michaelcushing) [inline src/pyspextool/extract/find_peaks.py:152]: Did something break here?  Why the float?

File: src/pyspextool/extract/find_peaks.py [hunk 4]
```diff
@@ -143,23 +143,25 @@ def find_peaks(profiles, method_info, fwhm=0.8):
             # Subtract the background
 
             y = y - np.median(y)
+            absy = np.abs(y)
```
Comment (michaelcushing) [inline src/pyspextool/extract/find_peaks.py:146]: Why this addition?  Was something not working?

File: src/pyspextool/extract/find_peaks.py (hunk 2 above)
Comment (michaelcushing) [reply src/pyspextool/extract/find_peaks.py:135]: Something is weird here as my version (and the main branch here) has some of these changes to find_peaks.py.  I am just confused as to why they are coming up again in a your pull request. @kelle .

**Comment by kelle** [comment]
I think this PR is making unnecessary changes to the test_data/ submodule. We don't need/shouldn't have the proc/ data in there.  Please run 
`git submodule update --init --recursive` 
to get the test_data/ directory back to it's original state.

**Review by kelle — APPROVED** [review]
